### ISSUE TITLE
Limit fields returned in Taxonomy Response without authentication.

### ISSUE THREAD

**Issue by rachelbaker** [WP-API/WP-API#824]
The default taxonomy response from `/wp/taxonomies` includes several fields that I believe we should not include without authentication. We also may want to consider requiring a context to include them as well.

These are:

- labels : suggestion **require auth** & consider exposing with a context
- types : suggestion **require auth**
- show_cloud : suggestion **require auth** & consider exposing with a context
- hierarchical : suggestion **require auth**

**Comment by danielbachhuber** [comment]
Why? Wouldn't you have access to labels in the admin even if you couldn't edit? And what does "edit" context even mean here anyway?

**Comment by rachelbaker** [comment]
Why give non-authenticated users these fields in the response?

**Comment by rmccue** [comment]
> Wouldn't you have access to labels in the admin even if you couldn't edit?

Not as far as I know. And not everyone has access to the admin at all in any case.

The advantage of putting these behind a context is that we can be explicit about errors. Only exposing them when `context=edit` allows us to give back a 403 error rather than just hiding/showing the fields (which could mean different things). `edit` might be the wrong name here, but it's at least consistent with other contexts.

**Comment by rachelbaker** [comment]
This comes down to two different issues:

1. We shouldn't expose object data that would normally not be accessible outside of wp-admin without at least requiring authentication. 

2. Is requiring authentication enough?  Or should we also additionally hide some behind a special context?  Where would all the taxonomy labels be useful?

**Comment by rachelbaker** [comment]
I updated the original issue text to clarify and add my 2 cents.

**Comment by danielbachhuber** [comment]
> edit might be the wrong name here, but it's at least consistent with other contexts.

Consistency through in consistency is not.

Worth saying I don't have a good suggestion yet. But I think it c

**Comment by danielbachhuber** [comment]
... could be more clear

**Comment by danielbachhuber** [comment]
Maybe the escalated context should be `admin`, `manage` or some equivalent name?

**Comment by TimothyBJacobs** [comment]
It could be very helpful in a front-end theming case. If you have a post type with a custom taxonomy, and you want to display all of the post objects terms, you'll need to pull a label of some kind.

**Comment by rachelbaker** [comment]
Wouldn't name or slug be sufficent for that case?

On Saturday, January 31, 2015, Timothy Jacobs <notifications@github.com>
wrote:

> It could be very helpful in a front-end theming case. If you have a post
> type with a custom taxonomy, and you want to display all of the post
> objects terms, you'll need to pull a label of some kind.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WP-API/WP-API/issues/824#issuecomment-72352127>.
>


-- 


Rachel Baker

**Comment by TimothyBJacobs** [comment]
Not necessarily, you'd really want the name, singular name for the CPT idea. But if you wanted to build a front end search by taxonomy. You'd need other things like search items, and no items found. Sounds a bit far-fetched, but I'm actually working on a project that works similar to this. We're not using the WP API for this, but we did use the registered taxonomy labels on the front-end, so we wouldn't have to hardcode the taxonomy to the view. We also did the same thing with if a taxonomy is hierarchal – it changed how we output our data.

Further, I don't really see any harm in any of that additional data.
